kep/594-resourcepolicy/README.md

```diff
@@ -0,0 +1,191 @@
+instances in my cluster. I hope that my workload can be deployed first on stable ECS instances, and during business peak 
+periods, the Pods that are scaled out are deployed on Spot instances. At the end of the business peak, the Pods on Spot 
+instances are prioritized to be scaled in.
+
+### Goals
+
+1. Develop a filter plugin to restrict the resource consumption on each unit for different workloads.
+2. Develop a score plugin to favor nodes matched by a high priority unit.
+3. Automatically setting deletion costs on Pods to control the scaling in sequence of workloads through a controller.
+
+### Non-Goals
+
+1. Modify the workload controller to support deletion costs. If the workload don't support deletion costs, scaling in 
+sequence will be random.
+2. When creating a ResourcePolicy, if the number of Pods has already violated the quantity constraint of the 
+ResourcePolicy, we will not attempt to delete the excess Pods.
+
+
+## Proposal
+
+### CRD API
+```yaml
+apiVersion: scheduling.sigs.x-k8s.io/v1alpha1
+kind: ResourcePolicy
+metadata:
+  name: xxx
+  namespace: xxx
+spec:
+  matchLabelKeys:
+    - pod-template-hash
+  matchPolicy:
+    ignoreTerminatingPod: true
+    ignorePreviousPod: false
+  forceMaxNum: false
+  podSelector:
+    matchExpressions:
+      - key: key1
+        operator: In
+        values:
+        - value1
+    matchLabels:
+      key1: value1
+  strategy: prefer
+  units:
+  - name: unit1
+    priority: 5
+    maxCount: 10
+    nodeSelector:
+      matchExpressions:
+      - key: key1
+        operator: In
+        values:
+        - value1
+  - name: unit2
+    priority: 5
+    maxCount: 10
+    nodeSelector:
+      matchExpressions:
+      - key: key1
+        operator: In
+        values:
+        - value2
+  - name: unit3
+    priority: 4
+    maxCount: 20
+    nodeSelector:
+      matchLabels:
+        key1: value3
+```
+
+`Priority` define the priority of each unit. Pods will be scheduled on units with a higher priority. 
+If all units have the same priority, resourcepolicy will only limit the max pod on these units.
+
+`Strategy` indicate how we treat the nodes doesn't match any unit. 
+If strategy is `required`, the pod can only be scheduled on nodes that match the units in resource policy. 
+If strategy is `prefer`, the pod can be scheduled on all nodes, these nodes not match the units will be 
+considered after all nodes match the units. So if the strategy is `required`, we will return `unschedulable` 
+for those nodes not match the units.
+
+`MatchLabelKeys` indicate how we group the pods matched by `podSelector` and `matchPolicy`, its behavior is like 
+`MatchLabelKeys` in `PodTopologySpread`.
+
+`matchPolicy` indicate if we should ignore some kind pods when calculate pods in certain unit.
+
+If `forceMaxNum` is set `true`, we will not try the next units when one unit is not full, this property have no effect
```

Comment:
Could you reword this sentence a bit?

Reply:
Rewrote this sentence. Pods will be matched by the ResourcePolicy in same namespace when the `.spec.podSelector`. And if `.spec.matchPolicy.ignoreTerminatingPod` is `true`, pods with Non-Zero `.spec.deletionTimestamp` will be ignored.